### PR TITLE
feat: Implement optional email results

### PR DESCRIPTION
**Pull request by uchajk**
I think this can be useful when export takes too long.

There is an optional email-to prop you can add to your button if you want the job to email the report once it's done.

### CODE REVIEW

**Comment by yajra** [comment]
The changes look good; I will review this as soon as I can. Thanks!

**Review by hpacleb — COMMENTED** [review]
Updating the README file on how to use this optional function would be appreciated

File: src/config/datatables-export.php [hunk 1]
```diff
@@ -48,6 +48,16 @@
     */
     's3_disk' => '',
 
+    /*
+    |--------------------------------------------------------------------------
+    | Mail from address
+    |--------------------------------------------------------------------------
+    |
+    | Will be used to email report from this address
```
Comment (hpacleb) [inline src/config/datatables-export.php:56]:
```suggestion
    | Will be used to email report from this address.
```

**Comment by yajra** [comment]
@hpacleb missing period suggestion merged. Do you approve this PR?

**Review by hpacleb — APPROVED** [review]
I'll just add the filename changes in another PR so this can be merged.

**Comment by challgren** [comment]
I'm going to add my 2 cents into this PR, I think users may want to customize the email template sent per datatable and not a generic one. But not really sure.

**Review by yajra — APPROVED** [review]

**Pull request merged**